src/Symfony/Component/DependencyInjection/Compiler/ResolveChildDefinitionsPass.php

```diff
@@ -137,7 +137,11 @@ private function doResolveDefinition(ChildDefinition $definition): Definition
             $def->setLazy($definition->isLazy());
         }
         if (isset($changes['deprecated'])) {
-            $def->setDeprecated($definition->isDeprecated(), $definition->getDeprecationMessage('%service_id%'));
+            if ($definition->isDeprecated()) {
+                $def->setDeprecation(...array_values($definition->getDeprecation('%service_id%')));
+            } else {
+                $def->setDeprecated(false);
```

Comment:
this should be removed: the method is going to be deprecated and this is not needed

Reply:
Yes, i'll do that, but I'm thinking how could this be replaced as with the new `setDeprecation` we cannot say that the service is not deprecated (I did'n think of that before)

The case when it's needed is when we have a deprecated service and we decorated it (and override the deprecation) https://github.com/symfony/symfony/blob/master/src/Symfony/Component/DependencyInjection/Tests/Compiler/ResolveChildDefinitionsPassTest.php#L311

WDYT?

Reply:
OK, let's keep the call here for BC, but this will be removed in v6. No need to keep this capability IMHO.